### PR TITLE
Interrupt assertion fix & more marchid-fixes

### PR DESCRIPTION
**Pull request by silabs-hfegran**

### CODE REVIEW

File: cv32e40x/tests/programs/custom/mhpmcounter29_csr_access_test_1/test.yaml [hunk 1]
```diff
@@ -4,6 +4,6 @@ description: >
     CSR access test with NUM_MHPMCOUNTER = 29 (can run with ISS)
 ovpsim: >
     --override root/cpu/misa_Extensions=0x001104
-    --override root/cpu/marchid=4
```
Comment (strichmo) [inline cv32e40x/tests/programs/custom/mhpmcounter29_csr_access_test_1/test.yaml:7]: Not anything wrong with your PR per se, but this diff reminds me that we should remove any marchid, misa overrides from test yamls.  Would you please remove both as part of this PR?

The marchid, misa extensions should flow from the test configuration YAMLs.

File: cv32e40x/tests/programs/custom/mhpmcounter29_csr_access_test_2/test.yaml [hunk 2]
```diff
@@ -4,6 +4,6 @@ description: >
     CSR access test with NUM_MHPMCOUNTER = 29 (CANNOT run with ISS)
 ovpsim: >
     --override root/cpu/misa_Extensions=0x001104
-    --override root/cpu/marchid=4
```
Comment (strichmo) [inline cv32e40x/tests/programs/custom/mhpmcounter29_csr_access_test_2/test.yaml:7]: Please remove misa and marchid.

**Review by strichmo — CHANGES_REQUESTED** [review]
Thanks @silabs-hfegran!  Only request is to remove a couple of superfluous ovpsim overrides.

File: cv32e40x/tests/programs/custom/mhpmcounter29_csr_access_test_1/test.yaml (hunk 1 above)
Comment (silabs-hfegran) [reply cv32e40x/tests/programs/custom/mhpmcounter29_csr_access_test_1/test.yaml:7]: Thanks, I'll remove those

**Review by strichmo — APPROVED** [review]

**Pull request merged**